lib/krb5/cache.c

```diff
@@ -443,10 +443,7 @@ environment_changed(krb5_context context)
 	 strncmp(context->default_cc_name, "API:", 4) == 0))
 	return 1;
 
-    if(issuid())
-	return 0;
-
-    e = getenv("KRB5CCNAME");
+    e = secure_getenv("KRB5CCNAME");
```

Comment:
This appears appears to be a bit of a change when suid, since (e == NULL) does not always just `return 0`.  Is that OK?